pkg/deployment/agent.go

```diff
@@ -28,17 +30,94 @@ func NewAgent(jaeger *v1alpha1.Jaeger) *Agent {
 }
 
 // Get returns a Agent pod
-func (a *Agent) Get() *appsv1.Deployment {
+func (a *Agent) Get() *appsv1.DaemonSet {
 	if strings.ToLower(a.jaeger.Spec.Agent.Strategy) != "daemonset" {
 		logrus.Infof(
 			"The Jaeger instance '%v' is using a Sidecar strategy for the Jaeger Agent. Skipping its DaemonSet deployment.",
-			a.jaeger.ObjectMeta.Name,
+			a.jaeger.Name,
 		)
 		return nil
 	}
 
-	logrus.Infof("DaemonSet deployments aren't supported yet. Jaeger instance: '%v'", a.jaeger.ObjectMeta.Name)
-	return nil
+	args := append(a.jaeger.Spec.Agent.Options.ToArgs(), fmt.Sprintf("--collector.host-port=%s:14267", service.GetNameForCollectorService(a.jaeger)))
+	trueVar := true
+	selector := a.selector()
+	annotations := map[string]string{
+		"prometheus.io/scrape": "true",
+		"prometheus.io/port":   "5778",
+	}
+
+	return &appsv1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "DaemonSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-agent-daemonset", a.jaeger.Name),
+			Namespace: a.jaeger.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				metav1.OwnerReference{
+					APIVersion: a.jaeger.APIVersion,
+					Kind:       a.jaeger.Kind,
+					Name:       a.jaeger.Name,
+					UID:        a.jaeger.UID,
+					Controller: &trueVar,
+				},
+			},
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: selector,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      selector,
+					Annotations: annotations,
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Image: a.jaeger.Spec.Agent.Image,
+						Name:  "jaeger-agent-daemonset",
+						Args:  args,
+						Ports: []v1.ContainerPort{
+							{
+								ContainerPort: 5775,
+								HostPort:      5775,
+								Name:          "zk-compact-trft",
```

Comment:
Is this populated to k8s objects? If so I would like to see the full names I had to think that means `zk-`

Reply:
It is, but unfortunately, there's a limit of 15 chars for this name. This is the best I could do with 15 chars while keeping it reasonably meaningful.

Reply:
that is weird IIRC in resource files we were using full names

Reply:
For *Container* ports? Or for service names? For the `ContainerPort` object, this is what I see in [Kubernetes docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#containerport-v1-core):

> If specified, this must be an IANA_SVC_NAME and unique within the pod. Each named port in a pod must have a unique name. Name for the port that can be referred to by services.

Where IANA_SVC_NAME is:

> at most 15 characters, matching regex [a-z0-9]([a-z0-9-]*[a-z0-9])* and it must containts at least one letter [a-z], hypens cannot be adjacent to other hyphens): e.g. "http"

Perhaps you are

Reply:
This https://github.com/jaegertracing/jaeger-kubernetes/blob/master/jaeger-production-template.yml#L78 it looks like a service port name